scripts/core/editor3/components/toolbar/InlineStyleButtons.tsx

```diff
@@ -42,14 +42,20 @@ export class InlineStyleButtonsComponent extends React.Component<any, any> {
 
     render() {
         const {editorFormat, editorState} = this.props;
-        const currentStyle = editorState.getCurrentInlineStyle();
+        const currentStyle = () => {
+            try {
```

Comment:
why do you use `try` here? Does `getCurrentInlineStyle` throw an error?

Reply:
Yes, it throws an error: 

```
Uncaught TypeError: Cannot read property 'getLength' of undefined
    at getInlineStyleForNonCollapsedSelection (EditorState.js:538)
    at EditorState.getCurrentInlineStyle (EditorState.js:211)
    at InlineStyleButtonsComponent.render (InlineStyleButtons.tsx:78)
    at finishClassComponent (react-dom.development.js:17039)
    at updateClassComponent (react-dom.development.js:16994)
    at beginWork$1 (react-dom.development.js:18505)
    at HTMLUnknownElement.callCallback (react-dom.development.js:347)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:397)
    at invokeGuardedCallback (react-dom.development.js:454)
    at beginWork$$1 (react-dom.development.js:23217)
    at performUnitOfWork (react-dom.development.js:22208)
    at workLoopSync (react-dom.development.js:22185)
    at renderRoot (react-dom.development.js:21878)
    at runRootCallback (react-dom.development.js:21554)
    at eval (react-dom.development.js:11353)
    at unstable_runWithPriority (scheduler.development.js:643)
```

so that's why I put a try catch block.

Reply:
Yes, it throws an error: 
```
Uncaught TypeError: Cannot read property 'getLength' of undefined
    at getInlineStyleForNonCollapsedSelection (EditorState.js:538)
    at EditorState.getCurrentInlineStyle (EditorState.js:211)
    at InlineStyleButtonsComponent.render (InlineStyleButtons.tsx:78)
    at finishClassComponent (react-dom.development.js:17039)
    at updateClassComponent (react-dom.development.js:16994)
    at beginWork$1 (react-dom.development.js:18505)
    at HTMLUnknownElement.callCallback (react-dom.development.js:347)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:397)
    at invokeGuardedCallback (react-dom.development.js:454)
    at beginWork$$1 (react-dom.development.js:23217)
    at performUnitOfWork (react-dom.development.js:22208)
    at workLoopSync (react-dom.development.js:22185)
    at renderRoot (react-dom.development.js:21878)
    at runRootCallback (react-dom.development.js:21554)
    at eval (react-dom.development.js:11353)
    at unstable_runWithPriority (scheduler.development.js:643)
```
so that's why I added a try and catch block.

Reply:
Did you find out the root cause of it? It would seem to me, an empty set would be returned if there are no inline styles.

Reply:
> an empty set

I thought the same but it seems like the problem is in DraftJS library, so I think It would be better, if we just put try and catch.

Reply:
An exception might also mean issues with editorState, so it'd be good to understand the reason for the exception since the same issue might also affect other parts of the code. Is it easy to reproduce? I could look into it as well.